x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/Role.java

```diff
@@ -93,6 +95,22 @@ public boolean checkIndicesAction(String action) {
         return indices().check(action);
     }
 
+
+    /**
+     * For given index patterns and index privileges determines allowed privileges and creates an instance of {@link ResourcesPrivileges}
+     * holding a map of resource to {@link ResourcePrivileges} where resource is index pattern and the map of index privilege to whether it
+     * is allowed or not.
+     *
+     * @param forIndexPatterns list of index patterns
+     * @param allowRestrictedIndices {@code true} whether restricted indices are allowed
+     * @param forPrivileges list of index privileges
+     * @return a map of resource to {@link ResourcePrivileges}
+     */
+    public ResourcesPrivileges getResourcePrivileges(List<String> forIndexPatterns, boolean allowRestrictedIndices,
```

Comment:
If this method is going to exist, then it needs `index` in its name somewhere, but I don't think it's really necessary is it?